app/controllers/api/v1/authentication_controller.rb

```diff
@@ -190,6 +186,31 @@ def auth_params
       params.require(:user_auth).permit(attributes)
     end
 
+    def valid_user
+      valid_mobile? && User.find_by_mobile(@mobile)
+    end
+
+    def valid_mobile?
+      User::HongKongMobileRegExp === @mobile
+    end
+
+    def render_invalid_mobile
+      attr = I18n.t('activerecord.attributes.user.mobile')
+      reason = @mobile.blank? ? 'blank' : (valid_mobile? ? 'not_registerd' : 'invalid')
+      err = I18n.t("activerecord.errors.models.user.attributes.mobile.#{reason}")
+      message = I18n.t('errors.format', attribute: attr, message: err)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.